src/commands/Pokemon/pokedex.ts

```diff
@@ -0,0 +1,176 @@
+import { toTitleCase } from '@klasa/utils';
+import { MessageEmbed } from 'discord.js';
+import { CommandStore, KlasaMessage } from 'klasa';
+import { SkyraCommand } from '../../lib/structures/SkyraCommand';
+import { getPokemonDetailsByFuzzy, GraphQLPokemonResponse, POKEMON_GRAPHQL_API_URL, POKEMON_EMBED_THUMBNAIL, parseBulbapediaURL } from '../../lib/util/Pokedex';
+import { fetch } from '../../lib/util/util';
+
+const BaseStats: Record<string, string> = {
+	hp: 'HP',
+	attack: 'ATK',
+	defense: 'DEF',
+	specialattack: 'SPA',
+	specialdefense: 'SPD',
+	speed: 'SPE'
+};
+
+export default class extends SkyraCommand {
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['pokemon', 'dex'],
+			cooldown: 10,
+			description: language => language.tget('COMMAND_POKEDEX_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_POKEDEX_EXTENDED'),
+			requiredPermissions: ['EMBED_LINKS'],
+			usage: '<pokemon:str>'
+		});
+	}
+
+	public async run(message: KlasaMessage, [pokemon]: [string]) {
+		try {
+			const embedTranslations = message.language.tget('COMMAND_POKEDEX_EMBED_DATA');
+			const { getPokemonDetailsByFuzzy: poke } = (await this.fetchAPI(message, pokemon)).data;
+
+			const baseStats: string[] = [];
+			let abilities = '';
+			let genders = '';
+			let evoChain = `**${toTitleCase(poke.species)}**`;
+
+			// Parse abilities
+			for (const [type, ability] of Object.entries(poke.abilities)) {
+				if (ability !== null) {
+					switch (type) {
+						case 'first':
+							abilities += ability;
+							break;
+						case 'hidden':
+							abilities += `, *${ability}*`;
+							break;
+						case 'second':
+						case 'special':
+							abilities += `, ${ability}`;
+					}
+				}
+			}
```

Comment:
For ease of maintenance, you should make abilities a `string[]`, which you later join by `', '`. This will make the code more readable and easier to maintain, plus you'll be able to convert the switch into a single if:

```typescript
for (const [type, ability] of Object.entries(poke.abilities)) {
	if (ability === null) continue;
	abilities.push(type === 'hidden' ? `*${ability}*` : ability);
}
```